### ISSUE TITLE
Compatibility with Dashware

### ISSUE THREAD

**Issue by DoofyOnABike** [Sublimis/UrbanBiker#168]
Hey!
I’ve been using Urban Biker to track my bike rides for over a year now and I couldn’t be any happier with the app. But I ran into an issue today that I feel could be fixed rather easily, by someone who actually knows what he/she is doing.
So, I just recently bought my very first GoPro and got into recording my trips. But since I am a huge fan of data and my Urban Biker is logging all sorts of fancy data (it’s paired with 2 extra sensors for cadence and speed), I was looking for a way to get this fancy data into my Videos.
Turns out it is very easy to actually achieve this with a freeware software called “Dashware”.
So, I imported one of my recent videos as well as the corresponding GPX-data that I had taken from Urban Biker. 
This entire process is rather simple and works great except for one tiny problem:
Dashware doesn’t seem to be able to find any values for the Cadence and is just showing “NaN” throughout the entire column even though the values are RIGHT THERE in the gpx-file.
After a long time of brainless and boring trial and error (I will spare you the details…), I actually found a solution for my problem!
Turns out that if I change this line in the exported GPX-file: ”xmlns:gpxtpx=http://www.garmin.com/xmlschemas/TrackPointExtension/v2”
To this: ”xmlns:gpxtpx=http://www.garmin.com/xmlschemas/TrackPointExtension/v1”
My problem is solved and I am getting a correct reading of all the data I wanted (speed, cadence, power, altitude, …) in Dashware.

Now I COULD open every single gpx-file that I export from UB and change that specific line, but that kinda feels like a dumb thing to do, since any “v2” should obviously be superior to its predecessor, shouldn’t it?? 
So long story short… Could you look into optimizing the compatibility of exported GPX-files with the software “Dashware”?

**Comment by Sublimis** [comment]
Hi!
Sorry, this cannot be fixed on our side.

Here is the probable common culprit: #https://github.com/tkrajina/gpxpy/issues/242.

It seems that both apps you mention use this library to read GPX files (just an educated guess), thus the same issue. Unfortunately, both apps appear to be abandoned as they haven't been updated for 4-5 years. Thus it's unlikely you'll see this fixed, even if the library does get fixed one day (as of 2022-03, the library still contains this issue).

We really do need the "v2" version of the track point extensions because v1 can't include speed data. 🤯  Please see for yourself: [view-source:https://www8.garmin.com/xmlschemas/TrackPointExtensionv2.xsd](https://www8.garmin.com/xmlschemas/TrackPointExtensionv2.xsd) versus [view-source:https://www8.garmin.com/xmlschemas/TrackPointExtensionv2.xsd](https://www8.garmin.com/xmlschemas/TrackPointExtensionv2.xsd).

There's a somewhat more clever thing you can do instead of opening every single file. Use a tool like [https://notepad-plus-plus.org/](https://notepad-plus-plus.org/) (or virtually any IDE environment) that has the feature called "Find in files" (or Replace in files). This way you can easily modify many GPX files within seconds.

Hope this helps! Thanks for the feedback! 🙂

**Comment by DoofyOnABike** [comment]
Hey!
Thanks for your answer, it was extremely helpful. I figured out a work around to solve my problem. Dashware can not only handle gpx-files but all sorts of raw data files so i used a little xslt to write all the values (Time,Latitude,Longitude,Elevation,Cadence,Speed,Power,Distance) that are found inside of the gpx-file into a csv-file. This CSV can then be used in Dashware making it irrelevant what libraries the software would otherwise use to read the gpx-files.
Which brings me to my next question: would it be possible for you to implement a way to export all the data that is usually stored in the gpx-file as a csv-file? This way I could just export the csv and woulnd’t have to deal with converting the gpx-files.
